src/tensors/tensor.jl

```diff
@@ -626,20 +440,31 @@ Return a view into the data slice of `t` corresponding to the splitting - fusion
 represents the slice of `block(t, c)` whose row indices correspond to `f₁.uncoupled` and
 column indices correspond to `f₂.uncoupled`.
 """
-@inline function Base.getindex(t::TensorMap{T,S,N₁,N₂,I},
+@inline function Base.getindex(t::TensorMap{T,S,N₁,N₂},
                                f₁::FusionTree{I,N₁},
                                f₂::FusionTree{I,N₂}) where {T,S,N₁,N₂,I<:Sector}
-    c = f₁.coupled
+    structure = tensorstructure(t)
     @boundscheck begin
-        c == f₂.coupled || throw(SectorMismatch())
-        haskey(t.rowr[c], f₁) || throw(SectorMismatch())
-        haskey(t.colr[c], f₂) || throw(SectorMismatch())
+        haskey(structure.fusiontreeindices, (f₁, f₂)) || throw(SectorMismatch())
     end
     @inbounds begin
+        i = structure.fusiontreeindices[(f₁, f₂)]
+        c = f₁.coupled
+        (d₁, d₂), r = structure.blockstructure[c]
+        r₁, r₂ = structure.fusiontreeranges[i]
         d = (dims(codomain(t), f₁.uncoupled)..., dims(domain(t), f₂.uncoupled)...)
-        return sreshape(StridedView(t.data[c])[t.rowr[c][f₁], t.colr[c][f₂]], d)
+        return sreshape(sreshape(sview(t.data, r), (d₁, d₂))[r₁, r₂], d)
```

Comment:
This could be slightly simplified by re-using the `block(t, c)` functionality. I am also slightly surprised by this, as I had expected the intermediate step of first unpacking the "matrixblock" and only then unpacking the "fusiontreeblock" could now be avoided by correctly adjusting the fusiontreeranges.

Reply:
Also I did not reuse `block(t,c)` since that uses `Base.view` and `reshape`, so then there is an extra conversion step into `StridedView`. Probably the same cost but yeah...